testnet/stacks-node/src/tests/integrations.rs

```diff
@@ -1025,6 +1025,172 @@ fn contract_stx_transfer() {
+                tenure
+                    .mem_pool
+                    .submit_raw(
+                        &mut chainstate_copy,
+                        &consensus_hash,
+                        &header_hash,
+                        xfer_to_contract,
+                    )
+                    .unwrap();
+            } else if round == 2 {
+                // block-height > 2
+                let publish_tx = make_contract_publish(&sk, 2, 0, "faucet", FAUCET_CONTRACT);
+                tenure
+                    .mem_pool
+                    .submit_raw(
+                        &mut chainstate_copy,
+                        &consensus_hash,
+                        &header_hash,
+                        publish_tx,
+                    )
+                    .unwrap();
+            } else if round == 3 {
+                let xfer_to_contract =
+                    make_stacks_transfer(&sk, 3, 0, &contract_identifier.into(), 1000);
+                tenure
+                    .mem_pool
+                    .submit_raw(
+                        &mut chainstate_copy,
+                        &consensus_hash,
+                        &header_hash,
+                        xfer_to_contract,
+                    )
+                    .unwrap();
+            } else if round == 4 {
+                let xfer_to_contract =
+                    make_stacks_transfer(&sk, 0, 0, &contract_identifier.into(), 1000);
+                tenure
+                    .mem_pool
+                    .submit_raw(
+                        &mut chainstate_copy,
+                        &consensus_hash,
+                        &header_hash,
+                        xfer_to_contract,
+                    )
+                    .unwrap();
+            }
+
+            return;
+        });
+
+    run_loop.callbacks.on_new_stacks_chain_state(
+        |round, _burnchain_tip, chain_tip, chain_state, burn_dbconn| {
+            let contract_identifier = QualifiedContractIdentifier::parse(&format!(
+                "{}.{}",
+                to_addr(&StacksPrivateKey::from_hex(SK_1).unwrap()).to_string(),
+                "faucet"
+            ))
+            .unwrap();
+
+            match round {
+                1 => {
+                    assert_eq!(chain_tip.metadata.block_height, 2);
+                    assert_eq!(chain_tip.block.txs.len(), 1);
+                }
+                2 => {
+                    assert_eq!(chain_tip.metadata.block_height, 3);
+                    assert_eq!(chain_tip.block.txs.len(), 1);
+                }
+                3 => {
+                    assert_eq!(chain_tip.metadata.block_height, 4);
+                    assert_eq!(chain_tip.block.txs.len(), 1);
+                }
+                4 => {
+                    assert_eq!(chain_tip.metadata.block_height, 5);
+                    assert_eq!(chain_tip.block.txs.len(), 2);
+
+                    // check that 1000 stx _was_ transfered to the contract principal
```

Comment:
I'm confused by this test.  I would have thought that all four transactions would have been mined into the same block, even though they were submitted out of order?  Why is it the case that the first STX-transfer (the one sent in round 4) is mined in a separate block from the three others?